File: ranker.pl
:- discontiguous pausable/2.
one_many(account,user).
one_many(directive,item).
one_many(account,item).
one_many(account,directive).
one_only(account).
once_only(account,user).
once_only(directive,item).
once_only(account,item).
once_only(account,directive).
grantable(account,user,item).
grantable(account,user,directive).
pausable(account,directive).
pausable(account,directive,item).
pausable(account,user).
pausable(account).

:- halt.

File: Ranker.hs
module Ranker where
import qualified Psql as S

relations = [
  e admin,e user,e item,e list,
  em admin,em user,
  t item "text",t list "directive",
  om admin user,om user list,om user item,
  m "comparison" [user,list,item,item]
  ]
  where
    admin = "admin"
    user = "user"
    item = "item"
    list = "list"
    e = S.entity
    m = S.mapping
    om = S.oneMany
    em = S.email
    t = S.text

ex = psql [S.entity "a",S.entity "b"]

file = readFile S.file >>= putStr
restart = psql [S.dropSchema schema,S.createSchema schema]
functions = psql [S.functions]
tables = psql [S.tables]
psql l = S.psql $ [S.changeSchema schema] ++ l

schema = "prioritizer"

notes = readFile "prioritizer.sql" >>= putStrLn

